python/feldera/pipeline.py

```diff
@@ -0,0 +1,46 @@
+from typing import Mapping, Any, Optional
+from feldera.attached_connector import AttachedConnector
+
+
+class Pipeline:
+    """
+    Represents a Feldera pipeline
+    """
+    id: Optional[str]
+    name: str
+    program_name: str
+    description: Optional[str]
+    attached_connectors: list[AttachedConnector]
+    config: Mapping[str, Any]
+    state: Optional[Mapping[str, Any]]
+    version: int
+
+    def __init__(
+        self,
+        name: str,
+        program_name: str,
+        description: Optional[str] = None,
+        attached_connectors: Optional[list[AttachedConnector]] = None,
+        config: Optional[Mapping[str, Any]] = None,

```

Comment:
Should there be a Configuration class which is just a wrapper around a map?

Reply:
I think, in the future, these can be proper classes.